crates/telio-wg/src/wg.rs

```diff
@@ -501,55 +569,92 @@ impl State {
     }
 
     fn time_since_last_rx(&self, public_key: PublicKey) -> Option<Duration> {
-        self.last_rx_timestamp
+        self.no_link_detection_info
             .get(&public_key)
-            .map(|v| Instant::now() - v.0)
+            .map(|p| p.last_rx_timestamp.elapsed())
+    }
+
+    async fn send_event(
+        &self,
+        state: PeerState,
+        link_state: Option<LinkState>,
+        peer: Peer,
+    ) -> Result<(), Error> {
+        // If the feature is disabled, the link_state field must be None
+        let link_state = if matches!(self.no_link_detection, NoLinkDetection::Disabled) {
+            None
+        } else if state != PeerState::Connected {
+            // If the peer is not connected, the link state must always be down
+            Some(LinkState::Down)
+        } else {
+            link_state
```

Comment:
Hmmm, curious if we need to pass `link_state` at all, judging from the glance at the code we can deduce it from `state` alone(conneted = link_state::up, else down)

Reply:
Consider if the peer state is connected and hinting mechanism detected possible link down. We want to send link down event.